registry/storage/driver/base/base.go

```diff
@@ -50,16 +50,40 @@ type Base struct {
 	storagedriver.StorageDriver
 }
 
+// Format errors received from the storage driver
+func (base *Base) setDriverName(e error) error {
+	if e != nil {
```

Comment:
go conventions are to return early and leave code as least indented as possible.  Could you invert this if statement?